firmware/common/usb_queue.c

```diff
@@ -26,6 +26,11 @@
 #include <stddef.h>
 #include <assert.h>
 
+// for __ldrex and __strex declarations
+#ifndef __ARM_ARCH_7M__
+	#define __ARM_ARCH_7M__
+#endif
+
```

Comment:
Again, this should be coming from the build system. In `firmware/hackrf-common.cmake` we have:
```cmake
SET(CPUFLAGS_M4 "-mthumb -mcpu=cortex-m4 -mfloat-abi=hard -mfpu=fpv4-sp-d16")
```
which should set this define as a side effect. If that's not being done in your build system then it's likely to cause other problems.

Reply:
Thanks for your comment. I removed the define.